src/Symfony/Component/DependencyInjection/ParameterBag/ContainerBagInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\ParameterBag;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface ContainerBagInterface extends ContainerInterface
```

Comment:
should we move this one namespace up? is typehinting `SF\ContainerBagInterface` instead of `SF\ParamaterBagInterface` really worth it? What about a `ContainerParameterBag` implem to enable typehinting `ParamaterBagInterface`?

Reply:
> should we move this one namespace up?

that's a parameter bag, should be in the namespace to me

> is typehinting SF\ContainerBagInterface instead of SF\ParamaterBagInterface really worth it?

yes: the new interface is more tailored than, the existing one, better for autosuggestion by IDEs

> What about a ContainerParameterBag implem to enable typehinting ParamaterBagInterface?

isn't that already what the new `ContainerBag` provides?

Reply:
>  the new interface is more tailored than, the existing one, better for autosuggestion by IDEs

Yeah but why favor `ContainerBagInterface` over `ParamaterBagInterface` from the same package/namespace? Is it to provide 2 different PSR containers? Are we "hacking" autowiring?

```
class ContainerParameterBag implements ParamBagInterface, PSR\ContainerInterface
{ }
```

would do?

Reply:
let's say you have `ContainerBagInterface $bag` type-hint.
typing `$bag->` the IDE will suggest you with "get", "has" and "all".
if we do what you suggest, the IDE will also suggest all the other methods inherited from `ParamaterBagInterface`, which, if used, will throw.

Reply:
Ok. I was thinking you'd prefer `ContainerInterface $bag` actually, which troubles autowiring when you need `$container` also. So we add this interface.. for DX. That's OK i guess :-)

Reply:
`ContainerBagInterface::resolveValue` would be useful?

Reply:
> ContainerBagInterface::resolveValue would be useful?

not for this PR

Reply:
> ContainerBagInterface::resolveValue would be useful

I changed my mind during the night, here it is, trivial to add :)

Reply:
Wait... do the point of this interface is just for DX? 🤔

Reply:
Is there anything we do in a framework that is not for DX?
Anyway, "for DX" is not a technical thing. Here is the technical outcome: the interface is to declare a contract you want to adhere to: you explicitly state that you're going to need read-only access to the bag. That's the purpose of the interface.